task-ballerina/src/task/listener.bal

```diff
@@ -22,18 +22,31 @@ public class Listener {
     *'object:Listener;
     boolean started = false;
 
-    private TimerConfiguration|AppointmentConfiguration listenerConfiguration;
+    private TaskConfiguration listenerConfiguration;
 
     # Initializes the `task:Listener` object. This may panic if the initialization is failed due to a configuration
     # error.
     #
-    # + configuration - The `task:TimerConfiguration` or `task:AppointmentConfiguration` record to define the
-    #   `task:Listener` behavior
-    public isolated function init(TimerConfiguration|AppointmentConfiguration configuration) {
-        if (configuration is TimerConfiguration) {
-            if (configuration["initialDelayInMillis"] == ()) {
-                configuration.initialDelayInMillis = configuration.intervalInMillis;
+    # + configuration - The configurations associated with the `task:Listener`
+    public isolated function init(TaskConfiguration configuration) {
```

Comment:
Rather have a new wrapper type called `TaskConfiguration`. shall we pass another param for `misfireConfiguration` in `init` method